log/httpLogger.go

```diff
@@ -0,0 +1,39 @@
+package log
+
+import (
+	"github.com/golang/glog"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+var _ retryablehttp.LeveledLogger = retryableHTTPLogger{}
```

Comment:
Why do we need this line? 🤔

Reply:
It confirms at compile time that the interface is satisfied, but I've realised it's not needed now I've added the constructor as that returns the interface and performs the same check at compile time.

Reply:
Then, I suggest to remove it :)